packages/blitz-auth/src/client/index.tsx

```diff
@@ -391,8 +397,55 @@ export const AuthClientPlugin = createClientPlugin((options: AuthPluginClientOpt
   globalThis.__BLITZ_SESSION_COOKIE_PREFIX = options.cookiePrefix || "blitz"
   return {
     withProvider: withBlitzAuthPlugin,
-    events: {},
-    middleware: {},
+    events: {
+      onRpcError: async (error: Error) => {
+        // We don't clear the publicDataStore for anonymous users,
+        // because there is not sensitive data
+        if (error.name === "AuthenticationError" && getPublicDataStore().getData().userId) {
+          getPublicDataStore().clear()
+        }
+      },
+      onSessionCreated: async (resetQueryClient) => {
+        resetQueryClient()
+        debug("Blitz Auth: onSessionCreated")
+      },
```

Comment:
This should be in blitz-rpc plugin. blitz-auth emits the onSessionCreated event, blitz-rpc responds to it